### PR TITLE
[Move] add script for tiered mint

### PR DESCRIPTION
**Pull request by moeziniaf**
Script for Treasury compliance account to invoke tiered mint and creation of DD

cargo test mint in transaction_builder/ tests

added some utils for non synthetic coins/cointypes

### CODE REVIEW

File: language/transaction-builder/src/lib.rs [hunk 1]
```diff
@@ -269,7 +269,7 @@ encode_txn_script! {
     name: encode_remove_validator_script,
     args: [to_remove: Address],
     script: RemoveValidator,
-    doc: "Encode a program adding `to_remove` to the set of pending validator removals. Fails if\
+    doc: "Encode a program adding `to_remove` to the set of pernding validator removals. Fails if\
```
Comment (dmitri-perelman) [inline language/transaction-builder/src/lib.rs:272]: lol, why do you like this spelling more?

File: language/stdlib/transaction_scripts/create_designated_dealer.move [hunk 2]
```diff
@@ -0,0 +1,8 @@
+script {
+    use 0x0::LibraAccount;
+        /// Script for Treasury Comliance Account to create designated dealer with 'new_account_address'
```
Comment (sblackshear) [inline language/stdlib/transaction_scripts/create_designated_dealer.move:3]:
```suggestion
        /// Script for Treasury Compliance Account to create designated dealer account at  'new_account_address'
```

File: language/transaction-builder/src/lib.rs [hunk 3]
```diff
@@ -446,6 +446,27 @@ encode_txn_script! {
...
+//...........................................................................
+// Treasury Compliance Scripts
+//...........................................................................
+
+encode_txn_script! {
+    name: encode_tiered_mint,
+    type_arg: coin_type,
+    args: [designated_dealer_address: Address, mint_amount: U64, tier_index: U64],
+    script: TieredMint,
+    doc: "Mints 'mint_amount' to 'designated_dealer_address' for 'tier_index' tier.\
```
Comment (sblackshear) [inline language/transaction-builder/src/lib.rs:458]: Nit: say what the max valid tier index is?

**Comment by dmitri-perelman** [comment]
I think bors doesn't understand `-r`, it should be `r=sblackshear `

**Pull request closed without merging**